lib/webdriveragent.js

```diff
@@ -80,12 +110,15 @@ class WebDriverAgent {
     if (this.webDriverAgentUrl) {
       log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
       this.url = this.webDriverAgentUrl;
+      this.setupNoSessionProxy();
       this.setupProxy(sessionId);
       return this.webDriverAgentUrl;
     }
 
     log.info('Launching WebDriverAgent on the device');
 
+    this.setupNoSessionProxy();
```

Comment:
We pass `sessionId` into the `launch` method. The `setupProxy` method is also called _before_ the `xcodebuild` command is run.